### PR TITLE
Referral Codes API for Stats

### LINKED ISSUE #1579: Add end-point to expose referral code channel data
GET api/v1/stats/referral_codes/BATS-123

```
{
  "promo_registration_id": "ABCDEFG12345",
  "channel": {
       "channel_id": "ABCDEFG12345",
       "channel_type" : "youtube",
       "channel_name": "Phil Defranco",
       "channel_identifier": "UClFSU9_bUb4Rc6OYfTt5SPw",
       "verified": true,
       "publisher_id": "ABCDEFG12345"
  }
```

### PR DESCRIPTION
**Pull request by dlipeles**
### Referral Codes API for stats

New routes at `/api/v1/stats/referral_codes` & `/api/v1/stats/referral_codes/:id`, closes #1579 

`/referral_codes`

```ruby
["BATS-123", "BATS-456", "BATS-789", ... ]
```

`/referral_codes/BATS-123`

```ruby
{
    "promo_registration_id":"441bfd72-be6d-4839-bb9a-55dccc89d4b1",
    "channel":
        {
          "channel_id"  :  "80500aeb-6198-4f88-bc8b-2e8a57c4e245",
          "channel_type"  :  "Youtube",
          "channel_name" : "Phil DeFranco",
          "channel_identifier" : "youtube#author:phil_defranco",
          "channel_stats" : "{ "Subscribers" :  "1234567", ...}",
          "channel_url" : "youtube.com/ABC1234567",
          "publisher_id" : "806fd992-025b-47af-adff-7412704da9a8",
          "verified" : true
         }
}
```

### CODE REVIEW

**Comment by dlipeles** [comment]
Still need to add unit tests, will add shortly.

**Comment by dlipeles** [comment]
Ready for review ✔️

File: app/controllers/api/v1/stats/referral_codes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class Api::V1::Stats::ReferralCodesController < Api::V1::StatsController
+    def index
+      data = PromoRegistration.all.map { |promo_registration| promo_registration.referral_code }
```
Comment (yachtcaptain23) [inline app/controllers/api/v1/stats/referral_codes_controller.rb]: Preferably use `pluck` as it's more readable and probably more performant.

e.g. `data = PromoRegistration.pluck(:referral_code)`

Comment (dlipeles) [reply app/controllers/api/v1/stats/referral_codes_controller.rb]: Read up on the doc, pluck is indeed a better method to use here. Thanks @yachtcaptain23

**Pull request merged**